optimum_benchmark/trackers/memory.py

```diff
@@ -39,19 +40,43 @@ def get_peak_memory(self):
         return bytes_to_mega_bytes(self.peak_memory)
 
     def _cuda_peak_memory(self):
-        import py3nvml.py3nvml as nvml
+        if is_nvidia_system():
+            if is_py3nvml_available():
+                import py3nvml.py3nvml as nvml
+            else:
+                raise ValueError(
+                    "The library py3nvml is required to run memory benchmark on NVIDIA GPUs, but is not installed. Please install it through `pip install py3nvml`."
+                )
+
+            handles = []
+            nvml.nvmlInit()
+            for device_index in self.device_ids:
+                handle = nvml.nvmlDeviceGetHandleByIndex(device_index)
+                handles.append(handle)
+            yield
+            for handle in handles:
+                meminfo = nvml.nvmlDeviceGetMemoryInfo(handle)
+                self.peak_memory += meminfo.used
+
+            nvml.nvmlShutdown()
+        elif is_rocm_system():
```

Comment:
thismethod is only called when `self.device.type == cuda`. Either we add a method `_rocm_peak_memory` or rename this one to `_gpu_peak_memory` and add `rocm` in the condition above.

Reply:
hmm unless torch sees `rocm` as `cuda` 🤔, is that the case ?

Reply:
yeah it is the case 😅 https://pytorch.org/docs/stable/notes/hip.html